behavior/plans/diff.py

```diff
@@ -1,35 +1,31 @@
 from __future__ import annotations
```

Comment:
General comment: In the near future, I would eventually like to take a pass on rewriting possibly most of this file to use vectorized pandas operations where possible, and `pandarellel` otherwise. The pipeline currently spends a significant portion of its time diffing on a single thread.

With that in mind, I just skimmed the changes. LGTM

Reply:
Yes.  The vast majority of dev time for this module was spent on analyzing and logging plan info and refactoring to use new TScout features.  There are definitely optimizations that can be done to speed up differencing.  I'm not sure to what extent the overhead is pandas copying dataframes vs pandas looping.  Either way, since the algorithm is stabilized now, this is a good candidate for optimization.